torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,221 @@
+        module.forward(output)
+
+def addObserver(myMod, qConfigDict, qConfigParent=None, prefix=''):
+    r"""Transform a module to a quantized module according to qConfigDict
+
+    This function insert observer module to all leaf child module of a given
+    module based on qConfigDict.
+
+    Args:
+        myMod: instance of the module we want to transform
+        qCofnigDict: dictionary that maps from name of submodule to quantization
+                     configuration
+        qConfigParent: quantization config of parent module, we will fallback to
+                       this config when there is no specified config for current
+                       module
+        prefix: corresponding prefix of the current module, used as key in
+                qConfigDict
+    """
+    myMod.qConfig = None
+    if prefix in qConfigDict:
+        myMod.qConfig = qConfigDict[prefix]
+    else:
+        if qConfigParent is not None:
+            myMod.qConfig = qConfigParent
+    # Insert observers only for leaf nodes
+    if myMod.qConfig is not None and len(myMod._modules) == 0:
+        myMod.add_module('observer', myMod.qConfig.activation())
+        myMod.register_forward_hook(_forward_hook)
+
+    for name, child in myMod.named_children():
+        if name is not 'observer':
+            if prefix:
+                module_prefix = prefix + '.' + name
+            else:
+                module_prefix = name
+            addObserver(child, qConfigDict, myMod.qConfig, module_prefix)
+
+def addQuantDeQuantCall(base_fn):
+    r"""Insert calls to quant and dequant around base_fn calls
+
+    Given a `base_fn` function, insert a call to `quant` before the function
+    call which is supposed to quantize the input and a call to `dequant` after
+    the function call which will dequantize the output of the function.
+
+    Args:
+        base_fn: base function we want to call
+
+    Returns:
+        new_fn: a modified function which calls quant -> base_bn -> dequant
+    """
+    def new_fn(x):
+        x = base_fn.__self__.quant(x)
+        x = base_fn(x)
+        x = base_fn.__self__.dequant(x)
+        return x
+    return new_fn
+
+def addQuantDeQuantModule(mod, qConfig):
+    r"""Add Quantize and DeQuantize to the input module and modify forward function
+
+    Args:
+        mod: module to be changed
+        qConfig: quantization configuration
+
+    """
+    mod.add_module('quant', AbstractQuant(qConfig))
+    mod.add_module('dequant', AbstractDeQuant())
+    mod.forward = addQuantDeQuantCall(mod.forward)
+
+def get_config_key(name, qConfigDict):
+    if name in qConfigDict:
+        return name
+    elif name:
+        parent = '.'.join(name.split('.')[:-1])
+        return get_config_key(parent, qConfigDict)
+    else:
+        return '' if '' in qConfigDict else None
+
+def getModule(modInstance, name):
+    splits = name.split('.')
+    mod = modInstance._modules[splits[0]]
+    for i in range(1, len(splits)):
```

Comment:
`for split in splits[1:]:`

and there's no reason to separate out the first iteration of the loop